packages/opencensus-instrumentation-http/src/http.ts

```diff
@@ -17,88 +17,121 @@
+   * @param moduleExports The http module exports
    * @param tracer A tracer instance to create spans on.
    * @param version The package version.
    */
-  // tslint:disable:no-any
-  applyPatch(moduleExporters: any, tracer: types.Tracer, version: string) {
-    this.setPluginContext(moduleExporters, tracer, version);
+  // disable-next-line to disable no-any check for moduleExporters parameter
+  // tslint:disable-next-line
+  applyPatch(moduleExports: any, tracer: types.Tracer, version: string) {
+    this.setPluginContext(moduleExports, tracer, version);
     this.logger = tracer.logger || logger.logger('debug');
 
     this.logger.debug('applying pacth to %s@%s', this.moduleName, this.version);
 
-    shimmer.wrap(moduleExporters, 'request', this.patchOutgoingRequest());
+    shimmer.wrap(moduleExports, 'request', this.patchOutgoingRequest());
 
     // In Node 8, http.get calls a private request method, therefore we patch it
     // here too.
     if (semver.satisfies(version, '>=8.0.0')) {
-      shimmer.wrap(moduleExporters, 'get', this.patchOutgoingRequest());
+      shimmer.wrap(moduleExports, 'get', this.patchOutgoingRequest());
     }
 
-    shimmer.wrap(
-        moduleExporters && moduleExporters.Server &&
-            moduleExporters.Server.prototype,
-        'emit', this.patchIncomingRequest());
+    if (moduleExports && moduleExports.Server &&
+        moduleExports.Server.prototype) {
+      shimmer.wrap(
+          moduleExports && moduleExports.Server &&
+              moduleExports.Server.prototype,
+          'emit', this.patchIncomingRequest());
+    } else {
+      this.logger.error(
+          'Could not apply patch to %s.emit. Interface is not as expected.',
+          this.moduleName);
+    }
 
-    return moduleExporters;
+    return moduleExports;
   }
 
+
   /** Unpatches all HTTP patched function. */
   applyUnpatch(): void {
-    shimmer.unwrap(this.moduleExporters, 'request');
+    shimmer.unwrap(this.moduleExports, 'request');
     if (semver.satisfies(this.version, '>=8.0.0')) {
-      shimmer.unwrap(this.moduleExporters, 'get');
+      shimmer.unwrap(this.moduleExports, 'get');
+    }
+    if (this.moduleExports && this.moduleExports.Server &&
+        this.moduleExports.Server.prototype) {
+      shimmer.unwrap(
+          this.moduleExports && this.moduleExports.Server &&
+              this.moduleExports.Server.prototype,
+          'emit');
     }
-    shimmer.unwrap(
-        this.moduleExporters && this.moduleExporters.Server &&
-            this.moduleExporters.Server.prototype,
-        'emit');
   }
 
   /**
    * Creates spans for incoming requests, restoring spans' context if applied.
    * @param plugin The HTTP plugin itself.
    */
   patchIncomingRequest() {
-    const plugin: HttpPlugin = this;
-    return (original: Function): Function => {
-      // tslint:disable:no-any
+    const plugin = this;
+    return (original: RequestFunction): types.Func<http.ClientRequest> => {
```

Comment:
The return type should be `RequestFunction` as well, right?

Reply:
Some side effects on the compilation. A TODO was added.